src/main/resources/schemas/workload.json

```diff
@@ -55,6 +55,16 @@
                         "title": "Time travel phase identifier",
                         "description": "If the task template supports time travel, this property can be used to specify the version of the data that will be queried by this task"
                       },
+                      "custom_task_executor": {
+                        "type": "string",
+                        "title": "Identifier for a custom task executor",
+                        "description": "If the task template should be executed with a custom task executor, this property can be used to specify which executor to use"
+                      },
+                      "dependent_batch_size": {
```

Comment:
Hmm, not sure I agree with that. The intention is different, right? Parameter values are values that are part of the SQL queries, the batch size is related to the algorithm instead. Thus, I'd argue they should be part of the instructions on how to execute the benchmark and not part of what to execute. I don't particularly like custom parameters to be specified here but we're doing it similarly for the TT id. Maybe it's something we should add as an issue?

Reply:
Yes, you are right. And it does not match what I mentioned in the other comment above, I think I just got confused :)

I think it should be in the workload indeed, but I think it should not be a first class citizen, since this is specific to a certain custom task implementation (that's what I had in mind). To make it generic, there could be a property `arguments` that accepts key-value pairs, such as `parameter_values` in an experiment definition. I think that change should be straightforward and will make the approach more generic and avoid introducing custom task specific properties in the workload schema.

Reply:
Added `arguments` to the experiment configuration and loaded the dependent batch size that way. I think this is definitely better than adding it to the task definition, what do you think?

Reply:
Then the problem is that you cannot have multiple batch sizes within same experiment? I was assuming the arguments were scoped to a custom task instantiation, i.e., similar to _arguments_ for a method in a programming language.

Reply:
This is an excerpt of how I thought the workload YAML file would look like:

```
...
- id: data_maintenance_1
  sessions:
  - tasks:
    - template_id: data_maintenance_delta_dependent
      custom_task_executor: com.microsoft.lst_bench.common.DependentTaskExecutor
      arguments:
        - delta_dependent: 10
    - template_id: data_maintenance_delta_dependent
      custom_task_executor: com.microsoft.lst_bench.common.DependentTaskExecutor
      arguments:
        - delta_dependent: 10
...
```

I think in one of my previous comments you mentioned you did not think arguments belonged to the experiment's config, and in fact, I agreed with that. It seems it's more natural that this is a way of parameterizing the task template instantiation when you are defining the workload. It also seems that introducing the arguments in the experiment's config, we need to bend a bit the spec by using the naming convention you were describing. What do you think?

Reply:
Okay, I think now I get what you're saying. Actually, what I was saying (or meant to say) in one of the earlier comments is that the batch size is a parameter of the task **execution** but not the task **definition**. Originally, the batch size was part of the task template definition, making it a static part of the definition of the workload rather than the a part that is dynamically defined as part of the execution, which is why I moved it into the experiment configuration. I think the workload configuration is a better alternative though. It's still structuring the execution but it's more task-oriented than the experiment configuration which makes sense. I've changed the above code snippet to use
`...
- id: data_maintenance_1
...
    - template_id: data_maintenance_delta_dependent
      custom_task_executor: com.microsoft.lst_bench.common.DependentTaskExecutor
      custom_task_executor_arguments:
        dependent_task_batch_size: 10
...`
instead.